### PR TITLE
Bug hunter

### PR DESCRIPTION
**Pull request by edsonarios**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 1]
```diff
@@ -13,6 +13,8 @@
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
+import static org.fundacion.jala.converter.core.parameter.Utils.cleanFileNameParameter;
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java:15]: delete blank line in line 15

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 2]
```diff
@@ -13,6 +13,8 @@
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java]: This line should be removed

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java [hunk 3]
```diff
@@ -22,6 +22,9 @@
 import org.springframework.web.multipart.MultipartFile;
 import java.io.File;
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: delete blank line in line 24

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java [hunk 4]
```diff
@@ -38,17 +41,17 @@ private ExtractFacade() {
      * @param  language is a type of language of the text.
      * @throws TextExtractorException if process is interrupted.
      */
-    public static void getTextExtract(final MultipartFile file, final String language) throws TextExtractorException {
+    public static void getTextExtract(final MultipartFile file, final String language) throws TextExtractorException, FileStorageException {
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: It pass 120 characters

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 3 above)
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: This line should removed

File: src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        for (int i = 0; i < fileName.length(); i++) {
+            if ((int) fileName.charAt(i) != asciiSpace) {
+                cleanedFileName += fileName.charAt(i);
+            }
+        }
+        return cleanedFileName;
+    }
+
+    /**
+     * Rename, removing whitespace in original file.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java]: Thirds person the verb the methods descriptions

File: src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java [hunk 6]
```diff
@@ -30,9 +30,12 @@
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java]: This line should be removed

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 3 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: check rest of class

File: src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java [hunk 7]
```diff
@@ -101,26 +105,29 @@ public AudioConverterInterface(final String newToken) {
...
-                int option = JOptionPane.showConfirmDialog(this, "File Path: " + file.getOriginFilePath()
-                        + "\nConvert to: " + audioSelect.getConvertTo() + "\nQuality: " + settings.getQuality()
-                        + "\nVolume: " + settings.getVolume() + "\nAudio Channel: " + settings.getAudioChannel()
-                        + "\nHz: " + settings.getHz() + "\nwith metadata: " + settings.isMetadata() + "\nChecksum: "
-                        + checksumLocal, "Message confirm", JOptionPane.YES_NO_OPTION);
-                if (option == 0) {
-                    callRequest();
-                } else {
-                    label.setVisible(false);
+            if(checkFormatAudioSupport(file.getOriginFilePath(), dotenv.get("AUDIO_FORMAT_SUPPORT"))) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java]: there must be a space after the if

File: src/main/java/org/fundacion/jala/converter/view/converter/ImageConverterInterface.java [hunk 8]
```diff
@@ -102,23 +103,25 @@ public ImageConverterInterface(final String newToken) {
...
-                checksumLocal = getFileChecksum(file.getOriginFilePath());
-                int option = JOptionPane.showConfirmDialog(this, "File Path: "
-                        + file.getOriginFilePath() + "\nConvert to: " + imageSelect.getConvertTo() + "\nWidth size: "
-                        + settings.getWidthSize() + "\nGray scale: " + settings.isGrayScale() + "\nChecksum: "
-                        + checksumLocal, "Message confirm", JOptionPane.YES_NO_OPTION);
-                if (option == NUMBER_ZERO) {
-                    callRequest();
-                } else {
-                    label.setVisible(false);
+            if(checkFormatAudioSupport(file.getOriginFilePath(), dotenv.get("IMAGE_FORMAT_SUPPORT"))) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/ImageConverterInterface.java]: There must be a space after the if

File: src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java [hunk 9]
```diff
@@ -30,9 +30,12 @@
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java]: This line should be  removed

File: src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java [hunk 10]
```diff
@@ -97,26 +100,28 @@ public VideoConverterInterface(final String newToken) {
...
-                        + "\nConvert to:" + menuConverterType.getConvertTo() + "\nResolutionWidth: "
-                        + settings.getWidthResolution() + "\nResolutionHeight: " + settings.getHeightResolution()
-                        + "\nFrames: " + settings.getFrame() + "\nSound: " + settings.isAudioSelected() + "\nThumbnail: "
-                        + settings.isThumbnailRequired() + "\nMetadata: " + settings.isMetadataRequired() + "\nChecksum: "
-                        + checksumLocal, "Message confirm", JOptionPane.YES_NO_OPTION);
-                if (option == 0) {
-                    callRequest();
-                } else {
-                    label.setVisible(false);
+            if(checkFormatAudioSupport(file.getOriginFilePath(), dotenv.get("VIDEO_FORMAT_SUPPORT"))) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java]: there must be a space after the if

File: src/main/java/org/fundacion/jala/converter/view/metadata/MetaDataInterface.java [hunk 11]
```diff
@@ -87,15 +88,19 @@ public void setPanels() {
...
-            JOptionPane.showMessageDialog(this, "File Path: " + file.getOriginFilePath()
-                    + "\nConvert to: " + exportingFormat.getConvertTo() + "\nMore information: "
-                    + exportingFormat.hasMoreInfo() + "\nOutput name: " + outputName.getOutPutName());
-            try {
-                LOGGER.info("Execute Try");
-                callRequest();
-            } catch (Exception ex) {
-                LOGGER.error("Execute Exception to metaData conversion");
-                ex.printStackTrace();
+            if(checkOutputNameMetadata(outputName.isSameName(), outputName.getOutPutName())) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/metadata/MetaDataInterface.java]: there must be a space after the if

File: src/main/java/org/fundacion/jala/converter/view/text_extractor/TextExtractorInterface.java [hunk 12]
```diff
@@ -83,16 +84,20 @@ public void setPanel() {
...
-                    + file.getOriginFilePath()
-                    + "\nConvert to: "
-                    + languageSelect.getConvertTo());
-            try {
-                LOGGER.info("Execute Try");
-                callRequest();
-            } catch (Exception ex) {
-                LOGGER.error("Execute Exception to text extraction");
-                ex.printStackTrace();
+            if(checkFormatAudioSupport(file.getOriginFilePath(), dotenv.get("TEXT_FORMAT_SUPPORT"))) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/text_extractor/TextExtractorInterface.java]: there must be a space after the if

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 13]
```diff
@@ -12,6 +12,8 @@
 
 import javax.swing.JOptionPane;
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: this line should be removed

File: src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java [hunk 14]
```diff
@@ -101,26 +105,29 @@ public AudioConverterInterface(final String newToken) {
...
+                } catch (ChecksumException checksumException) {
+                    checksumException.printStackTrace();
                 }
                 LOGGER.info("finish");
-            } catch (ChecksumException ioException) {
-                ioException.printStackTrace();
-                LOGGER.error("Execute Exception");
             }
         }
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java]: this linew should be removed

File: src/main/java/org/fundacion/jala/converter/core/facade/ChecksumFacade.java [hunk 15]
```diff
@@ -20,6 +20,8 @@
 import java.util.List;
 import java.util.stream.Collectors;
 import static org.fundacion.jala.converter.core.ChecksumService.getFileChecksum;
+import static org.fundacion.jala.converter.core.parameter.Utils.changeNameFile;
+import static org.fundacion.jala.converter.core.parameter.Utils.cleanFileNameParameter;
 import static org.fundacion.jala.converter.models.AssetSQL.*;
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/core/facade/ChecksumFacade.java]: Only import the necessary

File: src/main/java/org/fundacion/jala/converter/view/metadata/MetaDataInterface.java [hunk 16]
```diff
@@ -87,15 +88,19 @@ public void setPanels() {
...
+                int option = JOptionPane.showConfirmDialog(this, "File Path: " + file.getOriginFilePath()
+                        + "\nConvert to: " + exportingFormat.getConvertTo() + "\nMore information: "
+                        + exportingFormat.hasMoreInfo() + "\nOutput name: " + outputName.getOutPutName(), "Message confirm", JOptionPane.YES_NO_OPTION);
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/metadata/MetaDataInterface.java]: This line has more 120 character

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 17]
```diff
@@ -27,8 +28,56 @@ public static boolean checkFileSelect(final String file) {
...
+                messageError("You need to specify an output name ");
+                return false;
+            }
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Check if the format is supported.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: Change Check to Checks

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 18]
```diff
@@ -27,8 +28,56 @@ public static boolean checkFileSelect(final String file) {
...
+            if (arrayFormatSupports[i].equals(fileNameLocal)) {
+                return true;
+            }
+        }
+        messageError("Unsupported format ");
+        return false;
+    }
+
+    /**
+     * Show error message in UI.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: Change Show to Shows

**Review by jpgonzales27 — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 3 above)
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: this blank space should be removed

File: src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.fundacion.jala.converter.core.RunCommand;
+
+/**
+ * This class contains utilities for the core.
+ */
+public class Utils {
+
+    /**
+     * Clear the name of blanks
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java]: The java docs should be in third person

File: src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java (hunk 5 above)
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/core/parameter/Utils.java]: The java docs should be in third person

File: src/main/java/org/fundacion/jala/converter/view/converter/ImageConverterInterface.java [hunk 20]
```diff
@@ -102,23 +103,25 @@ public ImageConverterInterface(final String newToken) {
     public void actionPerformed(final ActionEvent e) {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/converter/ImageConverterInterface.java:103]: The java doc should be in third person

File: src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java (hunk 9 above)
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/converter/VideoConverterInterface.java]: THis blank space should be removed

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java (hunk 13 above)
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: This blank space should be removed

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 21]
```diff
@@ -27,8 +29,56 @@ public static boolean checkFileSelect(final String file) {
...
             return true;
         } else {
-            JOptionPane.showMessageDialog(null, "You need to select a file", "Error Message", JOptionPane.ERROR_MESSAGE);
+            messageError("You need to select a file");
             return false;
         }
     }
+
+    /**
+     * Verify that an output name was placed for the file.
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: The java docs should be in third person

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 22]
```diff
@@ -27,8 +29,56 @@ public static boolean checkFileSelect(final String file) {
...
+                messageError("You need to specify an output name ");
+                return false;
+            }
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Check if the format is supported.
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: The java docs should be in third person

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 23]
```diff
@@ -27,8 +29,56 @@ public static boolean checkFileSelect(final String file) {
...
+            if (arrayFormatSupports[i].equals(fileNameLocal)) {
+                return true;
+            }
+        }
+        messageError("Unsupported format ");
+        return false;
+    }
+
+    /**
+     * Show error message in UI.
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java]: The java docs should be in third person

**Review by GustavoZHA — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 24]
```diff
@@ -52,7 +54,7 @@ public static String getLinkExtractText(final MultipartFile file) {
      * @return a String of path which permit download the result process.
      */
     public static String getLinkMetadata(final String filename, final String format) {
-        return baseUrl + "/api/download/" + filename + "." + format;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java:55]: missing java doc

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 25]
```diff
@@ -42,7 +43,8 @@ public static String getLinkConverter(final String outputFilename) {
     public static String getLinkExtractText(final MultipartFile file) {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java:43]: The java docs in this class should be in third person

File: src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java [hunk 26]
```diff
@@ -27,8 +28,56 @@ public static boolean checkFileSelect(final String file) {
         if (file != null) {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/utilities/CheckFile.java:28]: The java docs should be in third person

**Review by danielaSantaCruz — APPROVED** [review]

**Pull request merged**